### PR TITLE
measure swap/allowance performance

### PR DESCRIPTION
**Pull request by gsteenkamp89**
- measure full running time of swap/allowance endpoint (add requests data as well as `crossSwapQuotes`).
- measure running time `getCrossSwapQuotes`, we can separate these by `amountType` eg, "exactInput" | "exactOutput".
- measure time to get crossSwapQuote by crossSwapType. eg "b2b" | "b2a" etc.

### CODE REVIEW

File: api/_dexes/cross-swap.ts [hunk 1]
```diff
@@ -78,6 +80,10 @@ export async function getCrossSwapQuotes(
 }
 
 export async function getCrossSwapQuotesForOutput(crossSwap: CrossSwap) {
+  const profiler = new Profiler({
+    at: "api/cross-swap#getCrossSwapQuotesForOutput",
+    logger: getLogger(),
```
Comment (dohaki) [inline api/_dexes/cross-swap.ts:85]: Do we need to specify some log level here?

**Review by dohaki — COMMENTED** [review]
Left a question

File: api/_dexes/cross-swap.ts (hunk 1 above)
Comment (gsteenkamp89) [reply api/_dexes/cross-swap.ts:85]: I think this logger only accepts a global log level.. I might have to change this

**Review by dohaki — APPROVED** [review]

File: api/_dexes/cross-swap.ts [hunk 2]
```diff
@@ -86,19 +91,35 @@ export async function getCrossSwapQuotesForOutput(crossSwap: CrossSwap) {
...
+    return await profiler.measureAsync(
+      getCrossSwapQuotesForOutputA2B(crossSwap),
+      "getCrossSwapQuotesForOutputA2B",
+      crossSwap
+    );
   }
 
   if (crossSwapType === CROSS_SWAP_TYPE.ANY_TO_ANY) {
-    return getCrossSwapQuotesForOutputA2A(crossSwap);
+    return await profiler.measureAsync(
```
Comment (james-a-morris) [inline api/_dexes/cross-swap.ts:118]: Do we need to run `return await`? You can usually omit the `await` because we're returning a promise anyway

**Review by james-a-morris — COMMENTED** [review]
one question

File: api/_dexes/cross-swap.ts (hunk 2 above)
Comment (gsteenkamp89) [reply api/_dexes/cross-swap.ts:118]: Yeah you're right this is not necessary!

**Pull request merged**